aioriak/tests/test_kv.py

```diff
@@ -69,6 +69,49 @@ def test_store_and_get(self):
             self.assertEqual(obj2.data, rand)
         self.loop.run_until_complete(go())
 
+    def test_store_with_links_with_explicit_tag(self):
+        async def go():
+            bucket = self.client.bucket(self.bucket_name)
+            rand = self.randint()
+            sub_obj = await bucket.new('sub', rand)
+            await sub_obj.store()
+
+            main_obj = await bucket.new('main', rand)
+            link_entry = (
```

Comment:
Maybe we should move `encode()` into `RiakPbcAsyncTransport._encode_content()`

```python
for link in robj.links:
    pb_link = rpb_content.links.add()
    try:
        bucket, key, tag = link
    except ValueError:
        raise RiakError("Invalid link tuple %s" % link)
    pb_link.bucket = bucket.encode()
    pb_link.key = key.encode()
    if tag:
         pb_link.tag = tag.encode()
    else:
        pb_link.tag = b''
```
And decode in `_decode_link()` method.
This changes broke backward compatibility but we can add bytes\str checking for the workaround this.
What do you think about it @jchristgit?

Reply:
That sounds like a good idea. I believe the `riak` library itself also uses regular strings in links instead of bytestrings. As for backwards compatibility, how does a helper function which shows a DeprecationWarning for passing bytes instead of strings and converts strings appropriately sound?

Reply:
Good idea! Can you fix this while I check CI?

Reply:
@singulared I've added link encoding to `_encode_content` and decoding to `_decode_link` and also added a test for the backwards-compatible behaviour (also making sure it raises a `DeprecationWarning`). As for CI, I've also discovered that the error Travis CI throws occurs when testing locally, on the first test run but then works on a second run. Maybe this is because RIak is still starting up and the command needs to have some retry logic?

Reply:
Nice, thanks! 
Yep, you are right. In some cases, Riak coordinator initialization may take more than 3 sec.
I think the simplest way to fix this is increasing sleep value:
https://github.com/rambler-digital-solutions/aioriak/commit/cf777b85deecdff569119ef145226adee14f9d08